lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -2720,16 +2722,75 @@ static llvm::Error checkAllNodesSupported(const Function &F, const Backend &B) {
   return llvm::Error::success();
 }
 
+/// Helper function that may transform \p F given preferences of \p cctx and
+/// \p B.
+static void transformForPrecisionMode(const Backend &B, Function *F,
+                                      const CompilationContext &cctx) {
+  const PrecisionConfiguration &precConfig = cctx.precisionConfig;
+
+  switch (precConfig.quantMode) {
+  case QuantizationMode::Profile: {
+    assert(cctx.bindings);
+    glow::profileQuantization(*cctx.bindings, F);
+    break;
+  }
+
+  case QuantizationMode::Quantize: {
+    quantization::quantizeFunction(F, precConfig.quantConfig, B,
+                                   *cctx.loweredInfoMap,
+                                   precConfig.precisionModeKindSet);
+    break;
+  }
+
+  case QuantizationMode::None: {
+    break;
+  }
+  }
+
+  if (precConfig.convertToFP16) {
+    TypeAToTypeBFunctionConverter converter(*F, ElemKind::FloatTy,
+                                            ElemKind::Float16Ty,
+                                            &precConfig.precisionModeKindSet);
+    converter.convert();
+  }
+}
+
 llvm::Error glow::optimizeFunction(Function *F, const Backend &B,
```

Comment:
Given that the documentation is very specific about these optimizations, maybe add a note here to update the documentation when this function changes